### DISCUSSION TITLE
Compose Desktop 打包出来的exe文件乱码

### DISCUSSION THREAD

**Discussion by doge-ak** [tangshimin/MuJing#36] [Q&A] [answered]
我在使用compose desktop这个框架的时候，出现了一点问题
```java
compose.desktop {
    application {
        mainClass = "org.example.antlr.MainKt"
        nativeDistributions {
            targetFormats(TargetFormat.Exe, TargetFormat.Msi)
            packageName = "C语言程序中的值传递"
            packageVersion = "1.0.0"
        }
    }
}
``` 
这个packageName  打包出来的文件名乱码，为      C璇█绋嬪簭涓殑鍊间紶閫抇

**Comment by doge-ak** [comment]
build.gradle.kts文件里的

**Reply by tangshimin** [reply]
你把 JDK 升级到 17 以上就可以使用中文程序名称了。可以选择 JDK 18 或者 JDK 21.

**Reply by tangshimin** [reply, marked as answer]
幕境之前使用 IBM 的 OpenJ9 打包后的程序，内存使用可以减少 100M 左右。
现在不推荐使用 OpenJ9 虚拟机了， Compose Desktop 程序推荐使用 jetbrains JDK。

**Reply by doge-ak** [reply]
已经解决了，谢谢大佬

**Reply by tangshimin** [reply]
不客气

**Comment by doge-ak** [comment]
我最近在看面试题的时候，终于知道为什么了。Java18开始默认字符集变成了utf8，我用的是Windows电脑。



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2023年11月12日 22:42 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [tangshimin/MuJing] 打包出来的exe文件乱码 (Discussion #36) |

不客气
